clippy_lints/src/unicode.rs

```diff
@@ -91,14 +91,14 @@ fn escape<T: Iterator<Item = char>>(s: T) -> String {
 
 fn check_str(cx: &LateContext<'_>, span: Span, id: HirId) {
     let string = snippet(cx, span, "");
-    if string.contains('\u{200B}') {
+    if let Some(invisible) = string.chars().find(|c| ['\u{200B}', '\u{ad}'].contains(&c)) {
         span_lint_and_sugg(
             cx,
             ZERO_WIDTH_SPACE,
             span,
-            "zero-width space detected",
+            &format!("invisible character detected: {:?}", invisible),
```

Comment:
As we are going to replace all occurrences, I think we should either specify all the characters that are going to be replaced, or just say `invisible character detected` without specifying which one (they will be visible in the suggestion anyway)

Reply:
Fair enough, thanks!